### PR TITLE
8273650: [lworld] Interpreter fails with fatal error: DEBUG MESSAGE: klass not initialized

### PR DESCRIPTION
**Pull request by fparain**
Please review those changes fixing handling of class initialization failures and adding unit tests to cover several failure scenarios and some corner cases (handling of instances from a failed initialization class).

Thank you,

Fred

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Issue
 * [JDK-8273650](https://bugs.openjdk.java.net/browse/JDK-8273650): [lworld] Interpreter fails with fatal error: DEBUG MESSAGE: klass not initialized

### Reviewers
 * [David Simms](https://openjdk.java.net/census#dsimms) (@MrSimms - Committer) ⚠️ Review applies to 728aaaba7615fba1060559767d628d00c4770795
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - Committer) ⚠️ Review applies to 728aaaba7615fba1060559767d628d00c4770795

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/valhalla pull/553/head:pull/553` \
`$ git checkout pull/553`

Update a local copy of the PR: \
`$ git checkout pull/553` \
`$ git pull https://git.openjdk.java.net/valhalla pull/553/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 553`

View PR using the GUI difftool: \
`$ git pr show -t 553`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/valhalla/pull/553.diff">https://git.openjdk.java.net/valhalla/pull/553.diff</a>

</details>

### CODE REVIEW

**Review by MrSimms — APPROVED** [review]
Looking good !

File: test/hotspot/jtreg/runtime/valhalla/inlinetypes/ClassInitializationFailuresTest.java [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+package runtime.valhalla.inlinetypes;
+
+
+import jdk.test.lib.Asserts;
+
+/*
+* @test
+* @summary Test several scenarios of class initialization failures
+* @library /test/lib
+* @run main/othervm  runtime.valhalla.inlinetypes.ClassInitializationFailuresTest
```
Comment (TobiHartmann) [inline test/hotspot/jtreg/runtime/valhalla/inlinetypes/ClassInitializationFailuresTest.java:32]: Just noticed that the `/othervm`  part can be removed because no flags are set. Also, the test should use 4-whitespace indentation.

File: test/hotspot/jtreg/runtime/valhalla/inlinetypes/ClassInitializationFailuresTest.java [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    static BadFour[] array;
+    static {
+      array = new BadFour[10];
+      if (ClassInitializationFailuresTest.failingInitialization) {
+        throw new RuntimeException("Failing initialization");
+      }
+    }
+  }
+
+  // Even if a primitive class fails to initialize property, some instances
```
Comment (TobiHartmann) [inline test/hotspot/jtreg/runtime/valhalla/inlinetypes/ClassInitializationFailuresTest.java:153]: "property" -> "properly"

File: test/hotspot/jtreg/runtime/valhalla/inlinetypes/ClassInitializationFailuresTest.java [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+  // Test handling of errors during the initialization of a primitive class
+  // Initialization of TestClass1 triggers the initialization of classes
+  // of all its primitive class typed fields, in this case BadOne
+  // Static initializer of BadOne throws an exception, so BadOne's initialization
+  // fails, which must caused the initialization of TestClass1 to fail too
+  // First attempt to instantiate TestClass1 must fail with an ExceptionInInitializerError
+  // because an exception has been thrown during the initialization process
+  // Second attempt to instantiate TestClass1 must fail with a NoClassDefFoundError
+  // because TestClass1 must already be in a faile initialization state (so no new
```
Comment (TobiHartmann) [inline test/hotspot/jtreg/runtime/valhalla/inlinetypes/ClassInitializationFailuresTest.java:61]: "faile" -> "failed"

**Review by TobiHartmann — APPROVED** [review]
Test looks good, just noticed some typos. I'll verify that it passes with the JIT before integrating JDK-8273594.

**Comment by fparain** [comment]
Thank you for the reviews.
All issues mentioned for the test file have been fixed.

Fred
